src/dotty/tools/dotc/ast/untpd.scala

```diff
@@ -94,17 +94,51 @@ object untpd extends Trees.Instance[Untyped] with UntypedTreeInfo {
   class InfixOpBlock(leftOperand: Tree, rightOp: Tree) extends Block(leftOperand :: Nil, rightOp)
 
   // ----- Modifiers -----------------------------------------------------
+  /** Mod is intended to record syntactic information about modifiers, it's
+    * NOT a replacement of flagsets.
+    *
+    * For any query about semantic information, check `flags` instead.
+    */
+  sealed trait Mod extends Positioned
 
- /** Modifiers and annotations for definitions
-   *  @param flags          The set flags
+  object Mod {
+    case class Private() extends Mod
```

Comment:
@odersky I just pushed a commit addressing the review feedback. Note that `addFlag` is also used for non-syntactic flags(such as `trait`, `method`), so I added another `addMod` method.

And it turns out that we always need to check the added flags, thus ` def | (mod: Mod) = ..` is not useful.